connection.go

```diff
@@ -28,6 +28,11 @@ const (
 	connClosed       = 2
 )
 
+const (
+	shutdownNotInProgress = 0
+	shutdownInProgress    = 1
+)
```

Comment:
I think it's time to rework the current state machine, perhaps add new states:

https://github.com/tarantool/go-tarantool/blob/265f96c9ff5f39e436e7645845fea4589762174d/connection.go#L25-L29

Reply:
Reworked